libraries/plugins/vfio/opae_vfio.c

```diff
@@ -1590,6 +1598,63 @@ fpga_result __VFIO_API__ vfio_fpgaGetIOAddress(fpga_handle handle,
 	return FPGA_OK;
 }
 
+fpga_result __VFIO_API__ vfio_fpgaBindSVA(fpga_handle handle, uint32_t *pasid)
+{
+	vfio_handle *h;
+	char path[PATH_MAX];
+	int fd;
+	int err;
+	fpga_result res = FPGA_OK;
+
+	h = handle_check_and_lock(handle);
+	ASSERT_NOT_NULL(h);
+	ASSERT_NOT_NULL(pasid);
+
+	struct opae_vfio *v = h->vfio_pair->device;
+	if (!v || !v->cont_pciaddr) {
+		res = FPGA_NO_DRIVER;
+		goto out_unlock;
+	}
+
+	if (h->pasid) {
+		// vfio_fpgaBindSVA() was already called. Return the same result.
+		*pasid = h->pasid;
+		goto out_unlock;
+	}
+
+	// Currently, the dfl-pci driver provides support for allocating a PASID
+	// and binding it to the FPGA in the IOMMU. At some point, VFIO and the
+	// IOMMUFD will also support shared virtual addressing. When ready, this
+	// function can be updated to call libopaevfio.
+
+	// dfl-pci creates a device at /dev/dfl-pci-sva/<pci_addr> for all
+	// ports that support shared virtual memory. If the device isn't found
+	// then assume PASID is not supported, either by the host or the FPGA.
+	snprintf(path, sizeof(path), "/dev/dfl-pci-sva/%s", v->cont_pciaddr);
+	fd = opae_open(path, O_RDONLY);
+	if (fd > 0) {
```

Comment:
```suggestion
	if (fd >= 0) {
```
0 is a valid file descriptor.